packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1843,13 +1843,13 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
   /// Returns positive if a is lower, negative if a is higher, 0 if their
   /// order can't be determine solely by their vertical position.
   static int _compareVertically(Rect a, Rect b) {
+    if ((a.top - b.top).abs() > _kSelectableVerticalComparingThreshold) {
```

Comment:
I noticed that on single-line text, the selectable fragment rects provided by `RenderParagraph` do not encompass each other even when split by `WidgetSpan`s. From my observation only on multi-line text is there some potential for encompassing rects.

This led me to move the rect top comparison first so that we don't immediately defer to horizontal comparison on multi-line text. Before this change we would immediately defer to horizontal comparison which is a little unreliable for encompassing rects that begin at the same `Offset`.

Reply:
This new logic fails on a single line text widget with a widget span, where the widget span has a greater height than the line.